consensus/src/experimental/tests/commit_phase_tests.rs

```diff
@@ -0,0 +1,695 @@
+                .await,
+            Err(Error::VerificationError)
+        );
+    });
+}
+
+#[test]
+fn test_commit_phase_check_commit() {
+    let mut runtime = consensus_runtime();
+    let (
+        _commit_tx,
+        _msg_tx,
+        _commit_result_rx,
+        _self_loop_rx,
+        _safety_rules_container,
+        signers,
+        _state_computer,
+        _validator,
+        mut commit_phase,
+    ) = prepare_commit_phase();
+
+    timed_block_on(&mut runtime, async move {
+        let signer = &signers[0];
+
+        let (vecblocks, li_sig) = prepare_executed_blocks_with_executed_ledger_info(signer);
+
+        assert!(commit_phase.blocks().is_none());
+
+        // when blocks is none
+        commit_phase.check_commit().await.ok();
+
+        assert!(commit_phase.blocks().is_none());
+
+        commit_phase.set_blocks(Some(PendingBlocks::new(vecblocks.clone(), li_sig.clone())));
+
+        // when blocks is good
+        commit_phase.check_commit().await.ok();
+
+        // the block should be consumed
+        assert!(commit_phase.blocks().is_none());
+        assert_eq!(commit_phase.load_back_pressure(), 1);
+
+        // when block contains bad signatures
+        let ledger_info_with_no_sig = LedgerInfoWithSignatures::new(
+            LedgerInfo::new(
+                vecblocks.last().unwrap().block_info(),
+                li_sig.ledger_info().consensus_data_hash(),
+            ),
+            BTreeMap::<AccountAddress, Ed25519Signature>::new(), //empty
+        );
+        commit_phase.set_blocks(Some(PendingBlocks::new(vecblocks, ledger_info_with_no_sig)));
+        commit_phase.check_commit().await.ok();
+
+        // the block should be there
+        assert!(commit_phase.blocks().is_some());
+    });
+}
+
+#[test]
+fn test_commit_phase_process_executed_blocks() {
+    let mut runtime = consensus_runtime();
+    let (
+        _commit_tx,
+        _msg_tx,
+        _commit_result_rx,
+        _self_loop_rx,
+        _safety_rules_container,
+        signers,
+        _state_computer,
+        _validator,
+        mut commit_phase,
+    ) = prepare_commit_phase();
+
+    timed_block_on(&mut runtime, async move {
+        let _signer = &signers[0];
+
+        let (vecblocks, li_sig) = prepare_executed_blocks_with_ordered_ledger_info(&signers[0]);
+
+        let ledger_info_with_no_sig = LedgerInfoWithSignatures::new(
+            LedgerInfo::new(
+                vecblocks.last().unwrap().block_info(),
+                li_sig.ledger_info().consensus_data_hash(),
+            ),
+            BTreeMap::<AccountAddress, Ed25519Signature>::new(), //empty
+        );
+
+        // no signatures
+        assert!(matches!(
+            commit_phase
+                .process_executed_blocks(vecblocks, ledger_info_with_no_sig)
+                .await,
+            Err(_),
+        ));
+    });
+}
+
+/// these functions are preserved to show the hierarchical structure of the tests.
+fn test_commit_phase_fn() {
```

Comment:
remove? we can organize the tests like

mod e2e_test {
  #[test]
  fn test_happy_path()
  #[test]
  fn test_bad_vote()  
}

mod functional_test() {
  #[test]
  fn test_process_commit_vote()
  ...
}

Reply:
resolved